include/seqan3/search/algorithm/detail/search.hpp

```diff
@@ -130,7 +130,7 @@ inline auto search_single(index_t const & index, query_t & query, configuration_
     }
     else
     {
-        using hit_t = std::conditional_t<index_t::is_collection_,
+        using hit_t = std::conditional_t<index_t::text_layout_mode,
```

Comment:
this is confusing, can you change it to `index_t::text_layout_mode == text_layout::collection`